pkg/gardenlet/controller/managedseed/managedseed_valueshelper.go

```diff
@@ -0,0 +1,362 @@
+			kubeconfig, err := ioutil.ReadFile(kubeconfigPath.(string))
+			if err != nil {
+				return nil, err
+			}
+			configValues, err = utils.SetToValuesMap(configValues, string(kubeconfig), "gardenClientConnection", "kubeconfig")
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	// If kubeconfig path is specified in seedClientConnection, read it and store its contents
+	kubeconfigPath, err := utils.GetFromValuesMap(configValues, "seedClientConnection", "kubeconfig")
+	if err != nil {
+		return nil, err
+	}
+	if kubeconfigPath != nil && kubeconfigPath.(string) != "" {
+		kubeconfig, err := ioutil.ReadFile(kubeconfigPath.(string))
+		if err != nil {
+			return nil, err
+		}
+		configValues, err = utils.SetToValuesMap(configValues, string(kubeconfig), "seedClientConnection", "kubeconfig")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Read server certificate file and store its contents
+	certPath, err := utils.GetFromValuesMap(configValues, "server", "https", "tls", "serverCertPath")
+	if err != nil {
+		return nil, err
+	}
+	if certPath != nil && certPath.(string) != "" && !strings.Contains(certPath.(string), secrets.TemporaryDirectoryForSelfGeneratedTLSCertificatesPattern) {
+		cert, err := ioutil.ReadFile(certPath.(string))
+		if err != nil {
+			return nil, err
+		}
+		configValues, err = utils.SetToValuesMap(configValues, string(cert), "server", "https", "tls", "crt")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Read server key file and store its contents
+	keyPath, err := utils.GetFromValuesMap(configValues, "server", "https", "tls", "serverKeyPath")
+	if err != nil {
+		return nil, err
+	}
+	if keyPath != nil && keyPath.(string) != "" && !strings.Contains(keyPath.(string), secrets.TemporaryDirectoryForSelfGeneratedTLSCertificatesPattern) {
+		key, err := ioutil.ReadFile(keyPath.(string))
+		if err != nil {
+			return nil, err
+		}
+		configValues, err = utils.SetToValuesMap(configValues, string(key), "server", "https", "tls", "key")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Delete server certificate and key paths
+	configValues, err = utils.DeleteFromValuesMap(configValues, "server", "https", "tls", "serverCertPath")
+	if err != nil {
+		return nil, err
+	}
+	configValues, err = utils.DeleteFromValuesMap(configValues, "server", "https", "tls", "serverKeyPath")
+	if err != nil {
+		return nil, err
+	}
+
+	return configValues, nil
+}
+
+func getParentGardenletDeployment(imageVector imagevector.ImageVector, shoot *gardencorev1beta1.Shoot) (*seedmanagementv1alpha1.GardenletDeployment, error) {
+	// Get image repository and tag
+	var imageRepository, imageTag string
+	gardenletImage, err := imageVector.FindImage("gardenlet")
+	if err != nil {
+		return nil, err
+	}
+	if gardenletImage.Tag != nil {
+		imageRepository = gardenletImage.Repository
+		imageTag = *gardenletImage.Tag
+	} else {
+		imageRepository = gardenletImage.String()
+		imageTag = version.Get().GitVersion
+	}
```

Comment:
I tend to disagree. From my perspective, conceptually `MergeWithParent=false` should mean that no parent values are used for anything, with all the resulting consequences, e.g. having to explicitly specify an image. We could add checks to the validation to ensure that this is the case.

The comparison with "before" is not meaningful - "before" there was no "merge with parent" option at all, instead the new gardenlet deployment was prepared based on an ad-hoc, inconsistent partial merge.

Reply:
After discussing this at length we agreed that `MergeWithParent` should only control whether the gardenlet configuration is merged or not. With respect to the deployment:
* a different image should be used if specified, or the parent one if not
* the parent image vector overwrites will right now always be used, since there is no way to specify them in the resource (but this could be introduced later)
* anything else is not merged with parent since the parent values are not even known (most of them could be known via the downward API but so far we didn't have a use case for this), so if a value is not specified explicitly a default one (usually an empty one) is used.

Reply:
Done